### ISSUE TITLE
Black screen on startup

### ISSUE THREAD

**Issue by fernie0341** [elishacloud/Silent-Hill-2-Enhancements#411]
Every time I try to run the game, my screen turns black and doesn't progress/becomes unresponsive (I have a 3990x Threadripper with a 3090 FTW3 GPU).

I followed the steps outlined in your installation guide, but I can't seem to get the game to run even once (full disclosure, I am using a download of Silent Hill 2: Restless Dreams from myabandonware.com). I have also tried running a vanilla version of the game using the No-CD executable from your site, but the game still crashes at the black screen. 

Here is the log file 
[d3d8.log](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6253961/d3d8.log)

**Comment by Polymega** [comment]
Thanks for the log file. We'll probably need @elishacloud's input on this, as it seems your log file has everything in order. The next steps that should happen in your log file would be:

```
Redirecting 'DirectSoundCreate8' ...
m_IDirectSound8::CreateSoundBuffer Created buffer!
```

**Comment by Polymega** [comment]
I just downloaded the "[fully extracted version](https://www.myabandonware.com/game/silent-hill-2-restless-dreams-bgd#download)" from MyAbandonware and it worked for me (after I swapped out the executable with the No-CD one). Did you download the "full extracted" one as well?

Right click `sh2pc.exe`, select **Properties**, go to the **Compatibility** tab, and make sure **all** compatibility options are unchecked/disabled.

**Comment by fernie0341** [comment]
Good morning,

I downloaded the fully extracted version and tried running it with the
No-CD, which yielded the following error (see attachment). Once I clicked
ok, the resolution on my desktop change (like 640-x640), wherin it gets
stuck on a black screen.

I doubled check the properties and everything is unchecked under the
compatibility tab.

Do you think it may have to do with a single-core affinity? When I checked
the affinity of sh2pc.exe in the details tab within task manager, I noticed
that I have an option to change processor groups (see attached image). I am
not sure if this is normal for most users or if this option is unique to my
computer since I am running a 3990x Threadripper.  When we specify the CPU
core in the "dsd8ini", does it consider what proccessor group the CPU core
is under?

Thanks for your help

Yes I extracted
On Apr 4, 2021, 11:33 AM -0700, Polymega ***@***.***>, wrote:

I just downloaded the "fully extracted version
<https://www.myabandonware.com/game/silent-hill-2-restless-dreams-bgd#download>"
from MyAbandonware and it worked for me (after I swapped out the executable
with the No-CD one). Did you download the "full extracted" one as well?

Right click sh2pc.exe, select *Properties*, go to the *Compatibility* tab,
and make sure *all* compatibility options are unchecked/disabled.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/411#issuecomment-813079633>,
or unsubscribe

**Comment by Polymega** [comment]
Hi there,

Attachments won't upload if you reply via email. Your response will need to be written through GitHub itself to upload attachments.

In regards to the Ryzen Threadripper grouping processor cores: I'm not sure if that would have any affect with running the game on a single core.

When I downloaded the "fully extracted version" from MyAbandonware just a moment ago, all I did was replace the executable with the "No-CD" one; I didn't use any of our project files. Meaning, when I ran my quick test, the game was using all available cores and it launched for me. (Of course, you don't want to actually play the game using multiple cores due to stability issues.)

You're right that if you can't get the base/vanilla game to launch for you, then it also won't launch with our project files. The base/vanilla game will always only launch in fullscreen mode between one of six pre-defined resolutions (with 640x480 being the default, I think).

------------------------

I was hoping some compatibility option was enabled through `sh2pc.exe` that just needed disabling. But here are some other random things to try:

- Make a copy of `sh2pc.exe` and rename it to something different such as `sh2pc_test.exe`. Launch the game with this newly named executable and see if that works.
- If you haven't tried so already, temporarily remove the DSOAL files and see if that works (`alsoft.ini`, `dsound.dll`, `dsoal-aldrv.dll`).
- I doubt this will do anything, but you can place these INI files in the game's directory and see if that fixes anything: [files.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6255190/files.zip)
   - These INI files are created for you after launching the game for the first time. But maybe they can help you if you already have them in the game's directory.

**Comment by fernie0341** [comment]
Good morning,

Thanks for letting me know about the attachments, I’ll make sure to attach them by replying in GitHub directly.

If you don’t think selecting a particular core in one processor group or another in task manager, that makes one less issue to worry about :)

The fact that I can’t get the game running using the vanilla game (with the no-cd executable) leads me to believe the crash is being caused by something intrinsic with my computer.

I will try the recommendations you listed. Should I try this on the base vanilla game or the modified game (game + enhance edition mods?

**Comment by Polymega** [comment]
> I will try the recommendations you listed. Should I try this on the base vanilla game or the modified game (game + enhance edition mods?

Preferably with just the base vanilla game. Although my second suggestion of removing the DSOAL files would be if you were testing with all of our other enhancement packages included.

**Comment by fernie0341** [comment]
Good afternoon,

I tried the steps that you recommended and it still doesn't work. It just
gets stuck on the black screen and the only way to close it is to use task
manager or restart the computer.

Regards,

Fernando Vargas

**Comment by fernie0341** [comment]
I can get the vanilla game running if I check compatibility mode for windows 95 or windows 98 under the compatibility tab in the properties folder of sh2pc.

**Comment by elishacloud** [comment]
Try the following:

1. Setting `CheckCompatibilityMode = 0` in d3d8.ini.
2. Adding the following line to the d3d8.ini file:  `SingleCoreAffinityTimer = 0`
3. Setting the game to run in compatibility mode for Windows XP.

**Comment by fernie0341** [comment]
Good evening,

I tried the suggestions that you recommended and the game is still getting
stuck on the black screen. I also tried changing the compatibility mode to
Windows 95 or 98 and launching the game and it fails to launch. The game
does launch if I have the compatibility set to Windows 98 or 98 and removed
the d3d8.dll file

Regards,

Fernando Vargas

**Comment by elishacloud** [comment]
Windows 95 or 98 compatibility mode is not needed for this game.  Also, the d3d8.dll was created with a newer version of Visual Studio and will never work with Windows 95 or 98 compatibility mode.  Try renaming `sh2pc.exe` to something else, like `sh2pc-new.exe` and then launching it.

**Comment by fernie0341** [comment]
Good afternoon,

Thanks for the info on the compatibility mode; I find it strange that the only way to play the vanilla game is while in windows 95 or 98 compatibility mode.

I tried changing the name to sh2pc.exe to sh2pc-new.exe and it still didn’t work. Let me know if you have any other recommendations.

Thanks for your help.

**Comment by elishacloud** [comment]
@fernie0341, thanks for testing this.  A few more things to try:

1. Check your video card settings (in the Nvidia Control Panel) and make sure to reset any settings for this game.
2. Try with this update and see if that helps: [d3d8.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6301162/d3d8.zip)

**Comment by fernie0341** [comment]
Good evening,

With respect to step 1, what exactly do you want me to reset?

Regards,

Fernando Vargas

**Comment by fernie0341** [comment]
I forgot to mention that I tried step two (after I reinstalled SH2 and the enhanced edition mods), and the game still gets stuck on the black screen. I included the log file in case you wanted it
[d3d8.log](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6301427/d3d8.log)

**Comment by elishacloud** [comment]
> With respect to step 1, what exactly do you want me to reset?

![image](https://user-images.githubusercontent.com/25625456/114502452-a8172200-9be0-11eb-9805-1deef99a76da.png)

You may need to do this for the "Global Settings" tab also.

In addition to resetting your Nvidia settings, can you try running the game with this launcher?  Just unzip this file into your game directory and double click it to launch the game.

Launcher:  [launch.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6301470/launch.zip)

**Comment by fernie0341** [comment]
Good evening,

I did not initially have sh2pc available to customize within Nvidia control panel, but I added left it as is. 

I extracted the launcher into the game and ran it but the game still gets stuck on the black screen. I have attached the log file for your reference.

Thanks for your help
[d3d8.log](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6301611/d3d8.log)

**Comment by fernie0341** [comment]
Good morning,

I wanted to inform you that the game is now working. I recently had to change the layout of my computer (finally got a waterblock for my GPU), which required me to restore the factory settings of my bios for testing;  I also removed the GEN 3 riser cable that I used to connect my GPU to the motherboard and I am now connecting my GPU directly to the motherboard.

I am not sure if the riser cable or the overclock settings I had on the bios was causing the game from launching, but after changing those two items the game is able to launch.

Thanks so much for your help and I look forward to playing the game with all the wonderful additions you and your community have added.

Regards
